### ISSUE TITLE
Standard RPCs should return structured responses

### ISSUE THREAD

**Issue by eddyashton** [microsoft/CCF#924]
A lot of our standard/common RPCs (`members/vote`, `mkSign`, etc) return naked bools as the response payload. This is awkward to parse, and the meaning is clear only by reading the handler's source code carefully. We should do a pass updating these to return structured JSON responses for machine-parsing, or potentially descriptive strings if we don't have any structured content (`Your vote for proposal 42 was accepted. The proposal has not passed.` rather than `true`).

**Comment by achamayou** [comment]
Do you mean a full object as opposed to a value? For `mkSign`, I'm not sure we even need a body, a return code seems enough.

Once we have a documented OpenAPI schema for these RPCs (eg. https://content-api.publishing.service.gov.uk/reference.html#content), they will become a lot more usable. I think this should be our more immediate direction, because it benefits all RPCs. Doing a pass over the common RPCs to make them more consistent is also good of course, but if they are at least systematically documented, it's already an improvement.

**Comment by eddyashton** [comment]
You're right about `mkSign`, that could just have no body. This was motivated by `members/vote`, where `{"completed": false}` would be better than `false`, and `{"your_current_vote": true, "proposal_id": 42, "completed": false}` would be even nicer.

With regards to OpenAPI, I think you're saying that we could add a description field saying "this bool value means X"? I agree that's an end goal, but in the shorter term named fields get us a lot of readability.

**Comment by achamayou** [comment]
Yes, exactly. I think it allows skipping long text descriptions for return codes, flags etc.

I agree that for votes, and some other RPCs, a little more detail wouldn’t hurt!
